packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -648,8 +649,32 @@ def cross_period_persisted_keys(self) -> List[str]:
         """Keys in the database which are persistent across periods."""
         return self._cross_period_persisted_keys.copy()
 
+    def initialize(self, synced_setup_data: Dict[str, List[Any]]) -> None:
+        """Initialize the data of the db.
+
+        This method should only be used to initialize the db, once,
+        and only after the agents have reached consensus about the initial setup data.
+        `RegistrationStartupRound` is meant to utilize this method.
+
+        :param synced_setup_data: the synchronized setup data to initialize the database with.
+        """
+        enforce(not self._initialized, "The database has already been initialized!")
+        enforce(
+            self.reset_index == RESET_COUNT_START,
+            "The database contains data before initialization!",
+        )
+        self._data = {
+            RESET_COUNT_START: synced_setup_data  # the key represents the reset index
+        }
```

Comment:
The db's data are first populated here, during the initialization, with synchronized setup data.